frontend/appflowy_flutter/lib/plugins/document/presentation/plugins/base/link_to_page_widget.dart

```diff
@@ -92,85 +93,167 @@ class LinkToPageMenu extends StatefulWidget {
 }
 
 class _LinkToPageMenuState extends State<LinkToPageMenu> {
+  final _focusNode = FocusNode(debugLabel: 'reference_list_widget');
   EditorStyle get style => widget.editorState.editorStyle;
+  int _selectedIndex = 0;
+  int _totalItems = 0;
+  List<dartz.Tuple2<AppPB, List<ViewPB>>> _availableLayout = [];
+  final Map<int, dartz.Tuple2<AppPB, ViewPB>> _items = {};
+  bool _isLoading = true;
+
+  Future<void> fetchItems() async {
+    final items = await AppBackendService().fetchViews(widget.layoutType);
+
+    setState(() {
+      _availableLayout = items;
+      _isLoading = false;
+    });
+
+    int index = 0;
+    for (final app in _availableLayout) {
+      for (final view in app.value2) {
+        _items.putIfAbsent(index, () => dartz.Tuple2(app.value1, view));
+        index += 1;
+      }
+    }
+
+    setState(() {
+      _totalItems = index;
+    });
+  }
```

Comment:
IIRC, we're adding to the list of items as long as it isn't a duplicate but we increment the index even if it's a duplicate. Should _totalItems = _items.length?

Reply:
Got it! Thanks.
P.S. putIfAbsent is just for precautionary purpose, logically there won't be duplicates, but yes I get your point. Thank you!

Reply:
I don't understand this piece of code. Why are you converting the list to a map?

Reply:
@LucasXu0 so the thing is that it is not list, it is a list of lists, but we need to consider it as one complete list, thus I am mapping indices to the object according to the one complete list we require. I am using map to finally insert the board/grid when user clicks enter. (I could have used a new list for this too, but choose map to do it, for no specific reason).

At the same time we also need list of lists for building the widget, as we need to separate the boards/grids belonging to different pages.